core/src/Lender.sol

```diff
@@ -133,27 +135,29 @@ contract Lender is Ledger {
     }
 
     function redeem(uint256 shares, address recipient, address owner) public returns (uint256 amount) {
-        // Guard against reentrancy, accrue interest, and update reserves
-        (Cache memory cache, uint256 inventory) = _load();
-
-        amount = _convertToAssets(shares, inventory, cache.totalSupply, /* roundUp: */ false);
-        require(amount != 0, "Aloe: zero impact");
-
         if (msg.sender != owner) {
             uint256 allowed = allowance[owner][msg.sender];
             if (allowed != type(uint256).max) allowance[owner][msg.sender] = allowed - shares;
         }
 
+        // Accrue interest and update reserves
+        (Cache memory cache, uint256 inventory) = _load();
+
+        // Convert `shares` to `amount`
+        amount = _convertToAssets(shares, inventory, cache.totalSupply, /* roundUp: */ false);
+        require(amount != 0, "Aloe: zero impact");
+
         // Burn shares, track rewards, and (if applicable) handle courier accounting
         cache.totalSupply = _burn(owner, shares, inventory, cache.totalSupply);
-
-        // Transfer tokens
+        // Assume tokens are transferred
         cache.lastBalance -= amount;
-        asset().safeTransfer(recipient, amount);
 
         // Save state to storage (thus far, only mappings have been updated, so we must address everything else)
         _save(cache, /* didChangeBorrowBase: */ false);
 
+        // Transfer tokens
+        asset().safeTransfer(recipient, amount);
```

Comment:
Similar to `deposit`, this is the only call in `redeem` that leaves the protocol. Interaction --> goes alst.